### PR TITLE
Provide option to control how <mi> is italicized.  (mathjax/MathJax#2570)

### PR DESCRIPTION
**Pull request by dpvc**
This PR adds an option to the TeX input jax that allows the author to control the `mathvariant` for Latin and Greek letters.  This is in response to a feature request from 2020.  You can specify one of four different math styles (`TeX`, `ISO`, `French`, and `upright`).  To do this, it introduces two new `ParseMethods` for upper- and lower-case Greek letters, and a `mathStyle` function in `ParseOptions` to get the variant to use for a given character.

Resolves issue mathjax/MathJax#2570.

### CODE REVIEW

File: ts/input/tex/ParseMethods.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ import {TexConstant} from './TexConstants.js';
 import {MmlNode} from '../../core/MmlTree/MmlNode.js';
 import ParseUtil from './ParseUtil.js';
 
+const MATHVARIANT = TexConstant.Variant;
```
Comment (zorkow) [inline ts/input/tex/ParseMethods.ts:32]: I believe we discussed getting rid of `TexConstant` once these PRs have been merged.

File: ts/input/tex/ParseOptions.ts [hunk 2]
```diff
@@ -29,15 +29,28 @@ import {NodeFactory} from './NodeFactory.js';
 import NodeUtil from './NodeUtil.js';
 import {MmlNode} from '../../core/MmlTree/MmlNode.js';
 import TexParser from './TexParser.js';
+import {TexConstant} from './TexConstants.js';
 import {defaultOptions, OptionList} from '../../util/Options.js';
 import {ParserConfiguration} from './Configuration.js';
 
+const MATHVARIANT = TexConstant.Variant;
+
```
Comment (zorkow) [inline ts/input/tex/ParseOptions.ts:37]: ditto + extra empty line.

File: ts/input/tex/ParseOptions.ts [hunk 3]
```diff
@@ -122,6 +140,7 @@ export default class ParseOptions {
     // Set default options for parser from packages and for tags.
     defaultOptions(this.options, ...options);
     defaultOptions(this.options, configuration.options);
+    this.mathStyle = ParseOptions.getVariant.get(this.options.mathStyle) || ParseOptions.getVariant.get('TeX');
```
Comment (zorkow) [inline ts/input/tex/ParseOptions.ts:143]: linelength

**Review by zorkow — APPROVED** [review]
Some minor things.

File: ts/input/tex/ParseMethods.ts (hunk 1 above)
Comment (dpvc) [reply ts/input/tex/ParseMethods.ts:32]: Right.

**Pull request merged**